nnstreamer_example/custom_example_opencv/nnstreamer_customfilter_opencv_average.cc

```diff
@@ -0,0 +1,144 @@
+/**
+ * NNStreamer OpenCV Custom Filter Example: Average
+ * Copyright (C) 2018 Sangjung Woo <sangjung.woo@samsung.com>
+ *
+ * LICENSE: LGPL-2.1
+ *
+ * @file  nnstreamer_customfilter_opencv_average.cc
+ * @date  25 Oct 2018
+ * @brief  OpenCV Custom NNStreamer Filter Example: Average
+ * @author  Sangjung Woo <sangjung.woo@samsung.com>
+ * @bug  No known bugs
+ * @see  nnstreamer_customfilter_example_average.c
+ * 
+ * This example calculates the average value of input tensor for 
+ * each channel (i.e. R, G & B). The shape of the input tensor is
+ * [N][y][x][M] and that of the output tensor is [N][1][1][M]. 
+ */
+
+#include "opencv2/opencv.hpp"
+
+#include <glib.h>
+#include <assert.h>
+#include <tensor_filter_custom.h>
+#include <tensor_common.h>
+
+/**
+ * @brief _pt_data Internal data structure
+ */
+typedef struct _pt_data
+{
+  uint32_t in_height;  /***< height of input tensor */
+  uint32_t in_width;   /***< width of input tensor */
+  uint32_t in_channel; /***< channel of input tensor */
+} pt_data;
+
+/**
+ * @brief pt_init
+ */
+static void *
+pt_init (const GstTensorFilterProperties * prop)
+{
+  pt_data *pdata = g_new(pt_data, 1);
+
+  return pdata;
+}
+
+/**
+ * @brief pt_exit
+ */
+static void
+pt_exit (void *private_data, const GstTensorFilterProperties * prop)
+{
+  pt_data *pdata = static_cast<pt_data *> (private_data);
+  assert (pdata);
+  g_free (pdata);
+}
+
+/**
+ * @brief set_inputDim
+ */
+static int
+set_inputDim (void *private_data, const GstTensorFilterProperties * prop,
+    const GstTensorsInfo * in_info, GstTensorsInfo * out_info)
+{
+  int i;
+  pt_data *pdata = static_cast<pt_data *> (private_data);
+
+  assert (pdata);
```

Comment:
If including glib in example app, please consider to use g_assert instead assert.